### PR TITLE
TreeView: Context Menu API

### LINKED ISSUE #5353: TreeView: Context Menu API

### PR DESCRIPTION
**Pull request by stsrki**
Closes #5353

I used the MouseEventArgs instead of BLMouseEventArgs because I think we must abandon BLMouseEventArgs in the long run.

### CODE REVIEW

File: Documentation/Blazorise.Docs/Pages/News/2024-02-15-release-notes-150.razor [hunk 1]
```diff
@@ -259,4 +259,12 @@
     In this release, we've updated our AntDesign provider. Previously, we were using the older 4.0 version of AntDesign, which didn't do justice to this exceptional framework. Although we aim to upgrade to the latest AntDesign 5.1 version in the upcoming Blazorise milestone, this update addresses several issues identified while developing Blazorise 1.5.
 </Paragraph>
 
+<Heading Size="HeadingSize.Is3">
+    TreeView context menu
+</Heading>
+
+<Paragraph>
+    It is now possible to detect when the user right click on a TreeView node. By using the <Code>NodeContextMenu</Code> event callback you can detect the clicked Node and mouse coordinates tos how your custom context menu.
```
Comment (David-Moreira) [inline Documentation/Blazorise.Docs/Pages/News/2024-02-15-release-notes-150.razor:267]: typo "tos how"

File: Demos/Blazorise.Demo/Pages/Tests/TreeViewPage.razor.cs [hunk 2]
```diff
@@ -129,4 +130,11 @@ protected IEnumerable<T> Flatten<T>( IEnumerable<T> source, Func<T, IEnumerable<
                 queue.Enqueue( child );
         }
     }
+
+    protected Task OnNodeContextMenu( TreeViewNodeMouseEventArgs<NodeInfo> eventArgs )
```
Comment (David-Moreira) [inline Demos/Blazorise.Demo/Pages/Tests/TreeViewPage.razor.cs:134]: Should we have an actual menu example in the docs?

**Review by David-Moreira — APPROVED** [review]
LGTM

**Comment by David-Moreira** [comment]
> Closes #5353
> 
> I used the MouseEventArgs instead of BLMouseEventArgs because I think we must abandon BLMouseEventArgs in the long run.

I agree. I don't see an advantage. Seems like the purpose is only to mirror the html events which the contextual blazor event args already do well so far.

File: Documentation/Blazorise.Docs/Pages/News/2024-02-15-release-notes-150.razor (hunk 1 above)
Comment (stsrki) [reply Documentation/Blazorise.Docs/Pages/News/2024-02-15-release-notes-150.razor:267]: This is just a template for me. I'm already working on rewriting release notes for another branch.

File: Demos/Blazorise.Demo/Pages/Tests/TreeViewPage.razor.cs (hunk 2 above)
Comment (stsrki) [reply Demos/Blazorise.Demo/Pages/Tests/TreeViewPage.razor.cs:134]: Added an example in the docs.

**Pull request merged**